src/control/server/mgmt_svc.go

```diff
@@ -850,11 +851,81 @@ func (svc *mgmtSvc) KillRanks(ctx context.Context, req *mgmtpb.RanksReq) (*mgmtp
 			rrErr = nil
 		}
 		resp.Results = append(resp.Results,
-			NewRankResult(i.getSuperblock().Rank.Uint32(), "stop",
-				state, rrErr))
+			NewRankResult(i.getSuperblock().Rank.Uint32(), "stop", state, rrErr))
+	}
+
+	svc.log.Debugf("MgmtSvc.KillRanks dispatch, resp:%+v\n", *resp)
+
+	return resp, nil
+}
+
+func ping(i *IOServerInstance, rank uint32, timeout time.Duration) *mgmtpb.RanksResp_RankResult {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	resChan := make(chan *mgmtpb.RanksResp_RankResult, 1)
+	go func() {
+		var err error
+		var dresp *drpc.Response
+
+		select {
+		default:
+			dresp, err = i.CallDrpc(drpc.ModuleMgmt, drpc.MethodPingRank, nil)
+		case <-ctx.Done():
+		}
+
+		resChan <- drespToRankResult(rank, "ping", dresp, err, system.MemberStateStarted)
```

Comment:
How does this actually play out if the context gets canceled? It seems like we should just abort (or respond that we aborted) rather than sending an empty result